pipeline.py

```diff
@@ -0,0 +1,202 @@
+     prev_coded_dir_path = args.prev_coded_dir_path
+
+     json_output_path = args.json_output_path
+     icr_output_dir = args.icr_output_dir
+     coded_dir_path = args.coded_dir_path
+     csv_by_message_output_path = args.csv_by_message_output_path
+     csv_by_individual_output_path = args.csv_by_individual_output_path
+     production_csv_output_path = args.production_csv_output_path
+
+     message_paths = [s01e01_input_path, s01e02_input_path, s01e03_input_path, s01e04_input_path, 
+                         s01e05_input_path, s01e06_input_path]
+     
+     # Load the pipeline configuration file
+     print("Loading Pipeline Configuration File...")
+     with open(pipeline_configuration_file_path) as f:
+          pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+
+     # Load phone number <-> UUID table
+     print("Loading Phone Number <-> UUID Table...")
+     with open(phone_number_uuid_table_path, "r") as f:
+          phone_number_uuid_table = PhoneNumberUuidTable.load(f)
+     
+          # Load messages
+     messages_datasets = []
+     for i, path in enumerate(message_paths):
+          print("Loading Episode {}/{}...".format(i + 1, len(message_paths)))
+          with open(path, "r") as f:
+               messages_datasets.append(TracedDataJsonIO.import_json_to_traced_data_iterable(f))
+
+     # Load surveys
+     print("Loading Demographics ...")
+     with open(s01_demog_input_path, "r") as f:
+          s01_demographics = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+     
+     print("Loading Follow up surveys ...")
+     with open(s01_follow_up_survey_input_path, "r") as f:
+          s01_follow_up_survey = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+     
+     # Add survey data to the messages
+     print("Combining Datasets...")
+     data = CombineRawDatasets.combine_raw_datasets(user, messages_datasets, [s01_demographics, s01_follow_up_survey])
+
+     print("Translating Rapid Pro Keys...")
+     data = TranslateRapidProKeys.translate_rapid_pro_keys(user, data, pipeline_configuration, prev_coded_dir_path)
+     
+     print("Writing TracedData to file...")
+     IOUtils.ensure_dirs_exist_for_file(json_output_path)
+     with open(json_output_path, "w") as f:
+          TracedDataJsonIO.export_traced_data_iterable_to_json(data, f, pretty_print=True)
+     
+     # Upload to Google Drive, if requested.
+     # Note: This should happen as late as possible in order to reduce the risk of the remainder of the pipeline failing
+     # after a Drive upload has occurred. Failures could result in inconsistent outputs or outputs with no
+     # traced data log.
+     if drive_upload:
+          print("Uploading CSVs to Google Drive...")
+
+          # Fetch the Rapid Pro Token from the Google Cloud Storage URL
+          parsed_rapid_pro_token_file_url = urlparse(pipeline_configuration.drive_credentials_file_url)
```

Comment:
On ADSS we recently moved the token download to happen first, so we can test the authentication before running the rest of the pipeline. Can you do that here too?

Reply:
To fix with generate analysis file pull